src/Core/CommandBus/MessengerCommandBusAdapter.php

```diff
@@ -0,0 +1,70 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShop\PrestaShop\Core\CommandBus;
+
+use Symfony\Component\Messenger\Exception\HandlerFailedException;
+use Symfony\Component\Messenger\Exception\LogicException;
+use Symfony\Component\Messenger\MessageBusInterface;
+use Symfony\Component\Messenger\Stamp\HandledStamp;
+
+/**
+ * Class MessengerCommandBusAdapter is the Symfony Messenger CommandsBus implementation for PrestaShop's contract.
+ */
+final class MessengerCommandBusAdapter implements CommandBusInterface
+{
+    public function __construct(private readonly MessageBusInterface $commandBus)
+    {
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function handle($command)
+    {
+        try {
```

Comment:
You should use `Symfony\Component\Messenger\HandleTrait` to avoid duplicating its code, now I know the two methods have the same name but @FabienPapet just taught me that you can rename the methods from traits.

You'll find an example here https://andy-carter.com/blog/overriding-extending-a-php-trait-method

```php
class Example
{
	use FooTrait {
		bar as traitBar;
	}

	public function bar()
	{
		return $this->traitBar() . 'B';
	}
}
```

Reply:
The injected MessageBusInterface is used in the HandleTrait

Reply:
I will post a comment to clarify this.

Reply:
Using "$this->messageBus->dispatch($command)" provides a Envelope object, not the expected result directly.
The trait directly retrieves the result and ensures that there are no multiple handlers that handled the result, in addition to other verifications. See https://symfony.com/doc/current/messenger/handler_results.html

Reply:
Does that mean we can't use middleware with this ?